Made convert tool use wildcard include exclude patterns

DreierF opened the pull request:
- [x] [No new Teamscale findings](https://build.cqse.eu/teamscale/delta.html#input/jacoco-client/?showMergeFindings=true)
- [ ] [Changes are tested adequately](https://build.cqse.eu/teamscale/tests.html#/jacoco-client/)
- [x] README.md updated in case of user-visible changes
- [x] CHANGELOG.md updated
- [x] build.gradle version number increased according to semantic versioning

salsolatragus commented on report-generator/src/main/java/com/teamscale/report/testwise/jacoco/cache/ProbesCache.java:
```diff
@@ -59,14 +60,16 @@ public boolean containsClassId(long classId) {
-							.getName() + " that was not provided. Either you did not provide " +
-							"all relevant class files or you did not adjust the include/exclude filters on the agent to exclude " +
-							"coverage from irrelevant code.");
+			if (locationIncludeFilter.test(executionData.getName())) {
```
I take that if `!containsClassId() && !locationIncludeFilter.test()` then the filter and the cache are in sync and everything is alright, therefore we don't need to log anything, right?

DreierF replied:
Exactly

salsolatragus commented on agent/src/main/java/com/teamscale/jacoco/agent/convert/ConvertCommand.java:
```diff
@@ -48,7 +48,7 @@
 	 * Ant-style exclude patterns to apply during JaCoCo's traversal of class files.
```
Comment mentions Ant-style.

salsolatragus commented on agent/src/main/java/com/teamscale/jacoco/agent/convert/ConvertCommand.java:
```diff
@@ -37,7 +37,7 @@
 	 * Ant-style include patterns to apply during JaCoCo's traversal of class files.
```
Comment mentions Ant-style.

DreierF replied:
🙈

DreierF commented on agent/README.md:
```diff
@@ -37,6 +37,9 @@ The following options are available:
   which are separated by line breaks. The file may also contain comments starting with `#`. (For details see path format)
+- `logging-config`: path to a [logback][] configuration XML file (other configuration formats are not supported at the moment).
+  Use this to change the logging behaviour of the agent. Some sample configurations are provided with the agent in the
+  `logging` folder, e.g. to enable debug logging or log directly to the console. (For details see path format)
```
Was already there, someone just forgot to link it together and I overlooked it while copy-pasting 😉

DreierF commented on agent/README.md:
```diff
@@ -144,6 +144,8 @@ The agent's REST API has the following endpoints:
 - `[GET] /test` Returns the name of the current test. The result will be empty when the test already finished or was 
   not started yet.
+  
+The uniformPath parameter must be url encoded if it may contain special characters (including /).
```
Good point

DreierF replied:
Also renamed uniformPath as it's something a non TS-Dev doesn't know

DreierF commented on agent/src/main/java/com/teamscale/jacoco/agent/testimpact/TestwiseCoverageAgent.java:
```diff
@@ -43,31 +43,46 @@ private void initServer() {
+			logger.error("End test name missing in " + request.url() + "!");
+
+			response.status(400);
+			response.body();
```
Should not have been there ;-)

DreierF commented:
We now use ant patterns for all file patterns and wildcard patterns for packages+classfiles.
Therefore the remaining occurrences are ok AFAIKS.
Good catch! Changed it to README.md and adjusted the template.
No clue what's wrong with the build 🤔 Worked now again
RFR

salsolatragus commented:
A side note: There doesn't seem to be coverage upload configured for this project. Should we remove the "adequately tested" item from the definition of done, because we cannot check it? Or should we configure coverage upload (probably the preferable way, I'm unaware of what this entails though).
